microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -0,0 +1,221 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from pathlib import PurePath
+
+from assertpy import assert_that
+
+from lisa import Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.features import Sriov
+from lisa.testsuite import simple_requirement
+from typing import List, Dict
+from lisa.util import SkippedException
+import itertools
+from .testpmd import Testpmd
+
+
+@TestSuiteMetadata(
+    area="dpdk",
+    category="functional",
+    description="""
+    This test suite check DPDK functionality
+    """,
+)
+class dpdk(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            This test case checks DPDK can be built and installed correctly.
+        """,
+        requirement=simple_requirement(
+            supported_features=[Sriov],
+        ),
+        priority=1,
+    )
+    def check_dpdk_build(self, node: Node) -> None:
+        sriov_feature = node.features[Sriov]
+        sriov_is_enabled = sriov_feature.enabled()
+        self.log.info(f"Verify SRIOV is enabled: {sriov_is_enabled}")
+        assert_that(sriov_is_enabled).is_true().described_as(
+            "SRIOV was not enabled for this test node."
+        )
+
+        self._hugepages_init(node)
+        self._hugepages_enable(node)
+
+        testpmd = Testpmd(node)
+        if not testpmd.install():
+            raise SkippedException()
+
+        node_nic_info = NodeNicInfo(node)
+
+        assert_that(len(node_nic_info)).described_as(
+            "Test needs at least 2 NICs on the test node."
+        ).is_greater_than_or_equal_to(2)
+
+        self.log.info(node_nic_info)
+
+        test_nic = node_nic_info.get_nic(node_nic_info.get_upper_nics()[-1])
+        vdev_type = "net_vdev_netvsc"
+        testpmd_include_str = test_nic.testpmd_include(vdev_type)
+        output = testpmd.run_with_timeout(testpmd_include_str, 30)
+        self.log.info(output)
+
+    def _execute_expect_zero(self, node: Node, cmd: str) -> str:
+        self.log.info(f"RUNCMD: {cmd}")
+        result = node.execute(cmd, sudo=True, shell=True)
+        assert_that(result.exit_code).described_as(
+            f"{cmd} failed with code {result.exit_code} and stdout+stderr:"
+            + f"\n{result.stdout}\n=============\n{result.stderr}\n=============\n"
+        ).is_zero()
+        if len(result.stdout) > 1024:
+            self.log.debug(
+                f"NOTE: Truncating result since output is larger than 1024.\n{result.stdout[:512]}\n.......(content has been truncated).......\n{result.stdout[-512:]}"
+            )
+        else:
+            self.log.debug(f"{cmd}:\n{result.stdout}")
+
+        return result.stdout
+
+    def _hugepages_init(self, node: Node) -> None:
+        self._execute_expect_zero(node, "mkdir -p /mnt/huge")
+        self._execute_expect_zero(node, "mkdir -p /mnt/huge-1G")
+        self._execute_expect_zero(node, "mount -t hugetlbfs nodev /mnt/huge")
```

Comment:
Try to leverage existing mount tool.

Reply:
You can add a parameter to support `-t` type.

Reply:
I didn't see a mount tool available when I looked, is it merged yet?

Reply:
It's in tools folder, and can be imported like `from lisa.tools import Mount`.